.github/workflows/update-self-version-reference.yml

```
@@ -0,0 +1,64 @@
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Get latest version
+        env:
+          GITHUB_TOKEN: ${{ secrets.ADGEM_SDK_ACTIONS_TOKEN }}
+          CHECK_TYPE: 'self'
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2181:style:2:6: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?

(shellcheck)